sympy/functions/special/error_functions.py

```diff
@@ -1414,6 +1414,18 @@ def _eval_rewrite_as_Si(self, nu, z, **kwargs):
     _eval_rewrite_as_Ci = _eval_rewrite_as_Si
     _eval_rewrite_as_Chi = _eval_rewrite_as_Si
     _eval_rewrite_as_Shi = _eval_rewrite_as_Si
+    
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.functions.special.gamma_functions import digamma
+        nu, z = self.args
+        if nu.is_real and nu >= 1:
+            term_one = (-z)**(nu - 1)/factorial(nu - 1) * (digamma(nu) - log(z))
+            term_two = 1/(nu - 1)
+            if nu == 1:
+                term_two = z/(2 - nu)
+            arg = Add(*[term_one, term_two]).as_leading_term(x, logx=logx)
+            return arg
+        return super(expint, self)._eval_as_leading_term(x, logx, cdir)
```

Comment:
Let's just use this PR to implement the _eval_as_leading_term function for the Si function. We shall look into other things later.